### PR TITLE
[spinel] first draft of I-D.rquattle-spinel-basis

### PR DESCRIPTION
**Pull request by jhwoodyatt**
This has loads of editor's notes that need to be cleared before we can submit to the draft queue.

### CODE REVIEW

**Comment by jwhui** [comment]
Please rebase to resolve conflicts.

**Comment by jhwoodyatt** [comment]
How does this look, @jwhui ?

**Comment by darconeous** [comment]
I assume this addresses #1472?

**Comment by jhwoodyatt** [comment]
Yes. This is to address #1472.

**Comment by darconeous** [comment]
The fact that this change duplicates lots of files makes it difficult to see if there were any significant changes, so I'll just ask: Did you make any significant changes to the files when you made the basis copies?

**Comment by darconeous** [comment]
And why copy? Why not just leave them (or just rename them) and remove their references from the  "unified" document?

I'm concerned because now it isn't clear which file is authoritative. Should people making changes update both files now? That seems really easy to mess up.

**Comment by jhwoodyatt** [comment]
p1. I made a lot of editorial changes, but the technical changes to the existing hierarchy of files was limited to removing the number reservations for things that seemed specific to Thread by my reading of the text. I'm in the middle of composing the files that will contain the definitions of those numbers in the unified draft that are missing from the basis.

p2. I copied the files rather than edit them in place out of a concern for keeping the unified draft from being changed by my introduction of the basis draft. I was planning to track any new changes to the unified draft as other contributors make them until the day arrives when we feel comfortable that the basis and the Thread specialization are a suitable replacement for it.

p3. I can see how that might lead to confusion with other contributors, so an alternative to consider is that perhaps I should push this change now into the separate spinel-documentation repository, and track the changes to the unified draft in the current openthread repository until it can be deleted from the openthread source tree. Let's talk about that.

**Comment by jwhui** [comment]
@darconeous @jhwoodyatt, is there a plan to move this forward?

**Comment by darconeous** [comment]
Now that I've got stuff I have been working on landed, this is my top next priority.

**Comment by darconeous** [comment]
Reviewing now.

**Comment by darconeous** [comment]
I'm going to go ahead and prepare this in a separate repository. Will update when I've got that all set up.

**Comment by darconeous** [comment]
See latest changes to https://github.com/openthread/spinel-spec

**Comment by Nopik** [comment]
Is something like encryption of the uart/spi traffic in the scope of this issue? I would love to see some proposals/comments about protecting spinel traffic from eavesdropping and/or other hacking behavior.

**Comment by darconeous** [comment]
@Nopik: That is is out of this document's scope, but we could write up another document defining a security layer for Spinel.

File: doc/spinel-protocol-src/spinel-basis-intro-encoding.md [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+ `s` | DATATYPE_INT16       | Signed 16-bit integer.
+ `L` | DATATYPE_UINT32      | Unsigned 32-bit integer.
+ `l` | DATATYPE_INT32       | Signed 32-bit integer.
+ `i` | DATATYPE_UINT_PACKED | Packed Unsigned Integer. See (#packed-unsigned-integer).
+ `6` | DATATYPE_IPv6ADDR    | IPv6 Address. (Big-endian)
+ `E` | DATATYPE_EUI64       | EUI-64 Address. (Big-endian)
+ `e` | DATATYPE_EUI48       | EUI-48 Address. (Big-endian)
+ `D` | DATATYPE_DATA        | Arbitrary data. See (#data-blobs).
+ `d` | DATATYPE_DATA_WLEN   | Arbitrary data with prepended length. See (#data-blobs).
+ `U` | DATATYPE_UTF8        | A text string encoded in [modified UTF-8][MUTF8] and terminated by U+0000 NUL character.
```
Comment (darconeous) [inline doc/spinel-protocol-src/spinel-basis-intro-encoding.md]: I just noticed this change. Changing this from UTF-8 to MUTF8 is a big change that is fundamentally incompatible with the current spinel parser, and seems unnecessary (especially replacing the null termination!). Why is this a good idea?

File: doc/spinel-protocol-src/spinel-basis-detail-status.md [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+113         | `STATUS_RESET_EXTERNAL`           | External device reset.
+114         | `STATUS_RESET_SOFTWARE`           | Internal software orderly reset.
+115         | `STATUS_RESET_FAULT`              | Internal software abortive reset.
+116         | `STATUS_RESET_CRASH`              | Unrecoverable software execution failure.
+117         | `STATUS_RESET_ASSERT`             | Software invariant property not respected.
+118         | `STATUS_RESET_OTHER`              | Unspecified cause.
+119         | `STATUS_RESET_UNKNOWN`            | Failure while recovering cause of reset.
+120         | `STATUS_RESET_WATCHDOG`           | Software failed to make sufficient progress.
+
+EDITOR: The `STATUS_CCA_FAILURE` status is a technology-specific status code. Should not be in the basis protocol.
```
Comment (darconeous) [inline doc/spinel-protocol-src/spinel-basis-detail-status.md]: `STATUS_CCA_FAILURE` doesn't seem technology-specific to me. You can have a CCA failure on any half-duplex medium.

**Comment by darconeous** [comment]
Closing. See https://github.com/openthread/spinel-spec

**Pull request closed without merging**